DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py

```diff
@@ -0,0 +1,274 @@
+import asyncio
+import pytest
+
+from dent_os_testbed.utils.test_utils.vlan_utils import (
+    configure_bridge_setup, configure_vlan_setup,
+    get_traffic_port_vlan_mapping)
+
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen,
+    tgen_utils_setup_streams, tgen_utils_start_traffic,
+    tgen_utils_stop_traffic, tgen_utils_get_traffic_stats,
+    tgen_utils_get_loss, tgen_utils_dev_groups_from_config,
+    tgen_utils_stop_protocols, tgen_utils_traffic_generator_connect,
+)
+
+pytestmark = pytest.mark.suite_functional_vlan
+
+packet_vids = ["X", 1, 2, 3, 4]  # VLAN tag number contained in the transmitted packet
+port_map = (
+    {"port": 0, "settings": [{"vlan": 2, "untagged": False, "pvid": False},
+                             {"vlan": 3, "untagged": True, "pvid": False},
+                             {"vlan": 4, "untagged": True, "pvid": True}]},
+    {"port": 1, "settings": [{"vlan": 2, "untagged": False, "pvid": False}]},
+    {"port": 2, "settings": [{"vlan": 3, "untagged": True, "pvid": True}]},
+    {"port": 3, "settings": [{"vlan": 4, "untagged": True, "pvid": True}]},
+)
+
+
+@pytest.mark.asyncio
+@pytest.mark.usefixtures("cleanup_bridges")
+async def test_vlan_all_supported_modes_broadcast(testbed):
+    """
+    Test Name: Broadcast with all modes supported
+    Test Suite: suite_functional_vlan
+ .   Test Overview: Test VLAN with broadcast when all modes supported
+    Test Procedure:
+    1.Initiate test params.
+    2Set links to vlans.
+    # 3.Map receiving and non receiving dut_ports.
+        Port 1 ->  tx_port
+        Ports 2, 3, 4 -> rx_ports
+    4.Setup packet stream(s) for the broadcast packet:
+    5.Send traffic to rx_ports
+         5.1 Verify traffic with vlan "x" received on port 4
```

Comment:
There is no VLAN X. But it could be "traffic without VLAN". Is this what you mean?